jormungandr/src/stats_counter.rs

```diff
@@ -39,11 +41,27 @@ impl StatsCounter {
             .fetch_add(count, Ordering::Relaxed);
     }
 
-    pub fn get_block_recv_cnt(&self) -> u64 {
+    pub fn block_recv_cnt(&self) -> u64 {
         self.stats.block_recv_cnt.load(Ordering::Relaxed) as u64
     }
 
-    pub fn get_uptime_sec(&self) -> u64 {
+    pub fn uptime_sec(&self) -> u64 {
         self.stats.start_time.elapsed().as_secs()
     }
+
+    pub fn set_slot_start_time(&self, time: SystemTime) {
+        self.stats
+            .slot_start_time
+            .write()
+            .unwrap_or_else(PoisonError::into_inner)
```

Comment:
Does this really need to survive a poisoned mutex?

Reply:
The mutex should never be poisoned, but even if it is, the underlying data is not corruptable with panics. Could be this or `.unwrap`, doesn't matter. I personally like avoiding statements capable of panicking.